services/worker/src/worker/job_runner.py

```diff
@@ -204,6 +204,7 @@ class JobRunner(ABC):
     worker_config: WorkerConfig
     common_config: CommonConfig
     processing_step: ProcessingStep
+    dataset_git_revision: Optional[str] = None
```

Comment:
nit: as this field is not meant to be accessed directly (it's a cache), I would prefer to prefix it with `_`
```suggestion
    _dataset_git_revision: Optional[str] = None
```